File: ElectronicBook.java
package org.example;

public class ElectronicBook extends Book {
    private final String format;
    private final long fileSizeInBytes;

    public ElectronicBook(String title, String author, String publisher, int year, String genre, String format, long fileSizeInBytes) {
        super(title, author, publisher, year, genre);
        this.format = format;
        this.fileSizeInBytes = fileSizeInBytes;
    }

    @Override
    public String getHeaders() {
        return super.getHeaders() + ",format,fileSizeInBytes";
    }

    @Override
    public String toCSV() {
        return super.toCSV() + "," + format + "," + fileSizeInBytes;
    }

}

File: ElectronicBook.py
from models.Book import Book

class ElectronicBook(Book):
    BYTES_PER_PAGE_COUNT = 2048

    def __init__(self, title, author, publisher, year, genre, file_format, file_size_in_bytes):
        super().__init__(title, author, publisher, year, genre)
        self.file_format = file_format
        self.file_size_in_bytes = file_size_in_bytes

    def get_pages_count(self):
        return self.file_size_in_bytes // self.BYTES_PER_PAGE_COUNT

    def do_something(self):
        return "ElectronicBook do_something result"
